frontend/src/store/cart/selectors.ts

```diff
@@ -0,0 +1,37 @@
+import { AppState } from "../rootReducer";
+import { getGameByIds } from '../catalogue/selectors';
+import { userGame } from './types';
+
+export const getCart = (state: AppState) => {
+  return state.cart.items;
+};
+export const getCartForUser = (state: AppState, currentCustomerId: string | undefined ) => {
+  const cart = getCart(state);
+  const userItems = cart.filter(({ customer_id }) => customer_id === currentCustomerId);
+
+  return userItems;
+}
+
+export const getUserItems = (state: AppState, currentCustomerId: string | undefined ) => {
+
+  const cart = getCart(state);
+  const userItems = cart.filter(({ customer_id }) => customer_id === currentCustomerId) || [];
+  const userProductsId = userItems.flatMap(({ items }) => (items || []).map(({ product_id }) => product_id));
+
+  const userGames = getGameByIds(state, userProductsId);
+  let gamesWithQuantity: userGame[] = [];
+
+  userGames.forEach((game) => { 
```

Comment:
This seams over complicated

Reply:
Agree but redux store structure turned out to be a little complicated. Everything is an array so we have to process the elements one by one in that way :(